test/common/testlib.py

```diff
@@ -1550,8 +1565,10 @@ def check_browser_errors(self):
                 raise Error(UNEXPECTED_MESSAGE + "browser errors:\n" + log)
 
     def check_pixel_tests(self):
-        if self.browser and self.browser.failed_pixel_tests > 0:
-            raise Error(PIXEL_TEST_MESSAGE)
+        if self.browser:
+            self.browser.assert_no_unused_pixel_test_references()
+            if self.browser.failed_pixel_tests > 0:
+                raise Error(PIXEL_TEST_MESSAGE)
```

Comment:
Nitpick: If there are failures, let's fail on these, and only then complain about unused refs. That makes typos in file names more obvious?

Reply:
Calling `assert_no_unused_pixel_test_references()` will not raise an exception itself, it will just increase `failed_pixel_tests`, in the same way as `assert_pixels`.  So we will always see all failures, including unused references.

No need to change anything here, IMO.

Reply:
I see, thanks!